src/main/java/org/broadinstitute/consent/http/service/DarCollectionService.java

```diff
@@ -144,4 +155,30 @@ public DarCollection updateDarCollection(Integer collectionId, User user) {
     return collections;
   }
 
+  // fetch elections in bulk via dar reference ids
+  // if any election exists, throw a bad request
+  // else collect the dar reference ids where the status is NOT cancelled
+  // once collected, perform bulk update where status is updated with cancelled
+  public DarCollection cancelDarCollection(DarCollection collection, User user) {
+    List<DataAccessRequest> dars = collection.getDars();
+    List<String> referenceIds = dars.stream()
+      .map(d -> d.getReferenceId())
+      .collect(Collectors.toList());
+    
+    List<Election> elections = electionDAO.findLastElectionsByReferenceIdsAndType(referenceIds, ElectionType.DATA_ACCESS.getValue());
+    if(!elections.isEmpty()) {
+      throw new BadRequestException("Elections present on DARs; cannot cancel collection");
+    }
+    List<String> nonCanceledIds = dars.stream()
+      .filter(d -> {
+        String status = d.getData().getStatus();
+        return Objects.nonNull(status) && status.toLowerCase() != "canceled";
+      })
+      .map(d -> d.getReferenceId())
+      .collect(Collectors.toList());
+    
+    dataAccessRequestDAO.cancelByReferenceIds(nonCanceledIds);
```

Comment:
Will need to catch the empty collection problem as this can throw a sql exception when that's true.